### PR TITLE
Spike: Refine the new proposal form submission

### PR DESCRIPTION
**Pull request by ken-futureverse**
## Summary

Update the form submission in NewProposal with a similar look and feel as `IdentityConnect` form

## Changes

- Add minor tweak to Button styling
- Remove the need to specify the second generic type `ISubmittableResult`
- Wrap custom input elements in `fowardRef`
- Implement a new `signAndSend` method with better error handling plus async/await friendly
- Added default function call to ensure the proposal is enacted correctly
- When pin data to Pinata, ensure it has a proper name, e.g `[rata] Proposal #51`

### CODE REVIEW

File: apps/gov-website/pages/proposal/new.tsx [hunk 1]
```diff
@@ -1,168 +1,86 @@
...
 		},
 	};
 };
 
-const NewProposal: NextPage<StaticProps> = ({ extrinsics }) => {
-	const { value: justification, onChange: onJustificationChange } =
-		useControlledInput<string, HTMLTextAreaElement>("");
+const NewProposal: NextPage<NewProposalProps> = ({ extrinsics }) => {
+	const { submitForm, formState } = useProposalNewForm();
+	const [formKey, setFormKey] = useState<string>(`PrposalNewForm${Date.now()}`);
```
Comment (aidan-starke) [inline apps/gov-website/pages/proposal/new.tsx:35]: React has a new hook `useId` for this purpose, though I'm not sure how you'd go about resetting it - worth a look

https://reactjs.org/docs/hooks-reference.html#useid

File: apps/gov-website/pages/proposal/new.tsx [hunk 2]
```diff
@@ -1,168 +1,86 @@
...
+					Submit a Proposal
+				</h1>
+
+				<p className="mb-8 text-lg">
+					To submit a proposal you must be a CENNZnet Councillor. Lorem laborum
+					dolor minim mollit eu reprehenderit culpa dolore labore dolor mollit
+					commodo do anim incididunt sunt id pariatur elit tempor nostrud nulla
+					eu proident ut id qui incididunt.
+				</p>
+				<FunctionCallFieldSet.Provider extrinsics={extrinsics}>
```
Comment (aidan-starke) [inline apps/gov-website/pages/proposal/new.tsx:74]: nitpick, but by wrapping the entire `ProposalNewForm` in this provider, every change to the call fields will cause a rerender of the whole form

File: libs/service/cennznet/src/findProposalId.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+): Promise<number | void> => {
+	const nextProposalId =
+		((await api.query.governance.nextProposalId())?.toJSON() as number) ?? 0;
+	const currentProposalId = Math.max(0, nextProposalId - 1);
+	const lastProposalId = Math.max(0, currentProposalId - FIND_DEPTH);
+
+	for (
+		let proposalId = currentProposalId;
+		proposalId >= lastProposalId;
+		proposalId--
```
Comment (aidan-starke) [inline libs/service/cennznet/src/findProposalId.ts:24]: fancy!

File: libs/web/hooks/src/useControlledInput.ts [hunk 4]
```diff
@@ -22,14 +22,14 @@ export const useControlledInput = <
 	};
 };
 
-export const useControlledCheckbox = (defaultValue: boolean) => {
-	const [value, setValue] = useState<boolean>(defaultValue);
+export const useControlledCheckbox = (defaultChecked: boolean) => {
+	const [checked, setCheked] = useState<boolean>(defaultChecked);
```
Comment (aidan-starke) [inline libs/web/hooks/src/useControlledInput.ts]: typo `setChecked`

**Review by aidan-starke — DISMISSED** [review]

File: apps/gov-website/pages/proposal/new.tsx (hunk 1 above)
Comment (ken-futureverse) [reply apps/gov-website/pages/proposal/new.tsx:35]: yeah good call, but looks like it's for assigning an id to element, not for key

File: apps/gov-website/pages/proposal/new.tsx (hunk 2 above)
Comment (ken-futureverse) [reply apps/gov-website/pages/proposal/new.tsx:74]: yeah you would have thought so eh, I checked and it doesn't seem to affect it for some reason. Try put in `console.log("render")` in both form and fieldset provider, only the provider is rerendered when you switch to different section/method. Can you check on your side before I merge this!?

File: libs/service/cennznet/src/findProposalId.ts (hunk 3 above)
Comment (ken-futureverse) [reply libs/service/cennznet/src/findProposalId.ts:24]: ✌

File: libs/web/hooks/src/useControlledInput.ts (hunk 4 above)
Comment (ken-futureverse) [reply libs/web/hooks/src/useControlledInput.ts]: gah

File: apps/gov-website/pages/proposal/new.tsx (hunk 2 above)
Comment (aidan-starke) [reply apps/gov-website/pages/proposal/new.tsx:74]: oh sweet yeah that checks out

**Review by aidan-starke — APPROVED** [review]

**Pull request merged**